python/langsmith/client.py

```diff
@@ -4664,6 +4664,538 @@ def _evaluate_strings(
+
+        Raises:
+            ValueError: If the current tenant is not the owner of the prompt.
+        """
+        owner, prompt_name, _ = ls_utils.parse_prompt_identifier(prompt_identifier)
+        if not self._current_tenant_is_owner(owner):
+            raise self._owner_conflict_error("delete a prompt", owner)
+
+        response = self.request_with_retries("DELETE", f"/repos/{owner}/{prompt_name}")
+
+        return response
+
+    def pull_prompt_object(
+        self,
+        prompt_identifier: str,
+        *,
+        include_model: Optional[bool] = False,
+    ) -> ls_schemas.PromptObject:
+        """Pull a prompt object from the LangSmith API.
+
+        Args:
+            prompt_identifier (str): The identifier of the prompt.
+
+        Returns:
+            ls_schemas.PromptObject: The prompt object.
+
+        Raises:
+            ValueError: If no commits are found for the prompt.
+        """
+        owner, prompt_name, commit_hash = ls_utils.parse_prompt_identifier(
+            prompt_identifier
+        )
+        use_optimization = ls_utils.is_version_greater_or_equal(
+            self.info.version, "0.5.23"
+        )
+
+        if not use_optimization and commit_hash == "latest":
+            latest_commit_hash = self._get_latest_commit_hash(f"{owner}/{prompt_name}")
+            if latest_commit_hash is None:
+                raise ValueError("No commits found")
+            else:
+                commit_hash = latest_commit_hash
+
+        response = self.request_with_retries(
+            "GET",
+            (
+                f"/commits/{owner}/{prompt_name}/{commit_hash}"
+                f"{'?include_model=true' if include_model else ''}"
+            ),
+        )
+        return ls_schemas.PromptObject(
+            **{"owner": owner, "repo": prompt_name, **response.json()}
+        )
+
+    def pull_prompt(
+        self, prompt_identifier: str, *, include_model: Optional[bool] = False
+    ) -> Any:
+        """Pull a prompt and return it as a LangChain PromptTemplate.
+
+        This method requires `langchain_core`.
+
+        Args:
+            prompt_identifier (str): The identifier of the prompt.
+
+        Returns:
+            Any: The prompt object in the specified format.
+        """
+        try:
+            from langchain_core.load.load import loads
+            from langchain_core.prompts import BasePromptTemplate
+        except ImportError:
+            raise ImportError(
+                "The client.pull_prompt function requires the langchain_core"
+                "package to run.\nInstall with `pip install langchain_core`"
+            )
+
+        prompt_object = self.pull_prompt_object(
+            prompt_identifier, include_model=include_model
+        )
+        prompt = loads(json.dumps(prompt_object.manifest))
+
+        if isinstance(prompt, BasePromptTemplate) or isinstance(
+            prompt.first, BasePromptTemplate
```

Comment:
Is there always going to be a `first` value if it's not a prompt template? (i.e., do we always enforce it'll be a runnable sequence then)?

Reply:
added better checking